### DISCUSSION TITLE
What timeout configuration exactly means?

### DISCUSSION THREAD

**Discussion by caibirdme** [apache/apisix#6116] [Q&A]
I found 3 timeout configs in apisix and in nginx:
1. connect_timeout
2. send_timeout
3. read_timeout

I searched for a while but cannot find clear answer, so I want to ask there:
for connect_timeout. it's intuitive( TCP dial timeout). But what's send_timeout and read_timeout?
from [nginx's doc](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_send_timeout), it means `the timeout between to successive send operatiosn`. And read_timeout means `the timeout between to successive read operatiosn`. But it's pretty confusing! I mean why I need to care about timeout between two isolated I/O operations? I just care about how much time the proxy have to wait for the upstream to handle the request and reponse to proxy. And I want to know how to configure for that purpose

**Comment by tokers** [comment]
The time interval for a proxy can send/read packets successfully usually can measure the network quality between the proxy and the backend, so that's why the meanings of these two timeouts will care about the successive operations. The total time for a request/response cannot be decided easily as it's subjected to the size of request and response.

**Reply by WeCreos** [reply]
What's the time interval for the backend?

**Reply by tokers** [reply]
@WeCreos It means how long will it take to complete two successive read/write operations.
